src/lwc/geFormRenderer/geFormRendererHelper.js

```diff
@@ -0,0 +1,91 @@
+import GeFormService from 'c/geFormService';
+import {isEmptyObject, isNotEmpty} from 'c/utilCommon';
+
+/**
+ * @description Helper function used to convert an object that has key value pairs where
+ * the value is an object with a value property, i.e. {value: {value:'', displayValue:''}},
+ * into an object that has primitives as values: {value: ''}.
+ * @param obj The object that has value objects.
+ * @returns An object that has primitives as values, derived from the "value" property of
+ * the fields on the passed in object.
+ */
+export function flatten(obj) {
+    let flatObj = {};
+    for (const [key, value] of Object.entries(obj)) {
+        if (value !== null && value !== undefined && value.hasOwnProperty('value')) {
+            flatObj[key] = value.value;
+        } else {
+            flatObj[key] = value;
+        }
+    }
+    return flatObj;
+}
+
+/**
+ * @description Helper function used to convert the additional object JSON format used by BDI to process
+ * additional objects for a donation into a JSON format used by widgets. This widget JSON format is also stord in
+ * the form renderer form state.
+ * @param additionalObjectJson The additional object JSON string that needs to be converted to the widget format
+ * @return {string} The newly converted JSON string that can be used by widgets to read additional objects
+ * applicable to the widget.
+ */
+export function convertBDIToWidgetJson(additionalObjectJson) {
+    if (isEmptyObject(additionalObjectJson)) {
+        return;
+    }
+
+    const additionalObjects = JSON.parse(additionalObjectJson);
```

Comment:
@dfuller114 What do you think about wrapping this call to JSON.parse in a try/catch?  In my scratch org I am seeing that anything not parseable in that field results in an unhandled exception.  I wonder if it might be better to catch and throw an Exception with a message along the lines of "Malformed Additional Object JSON value..."  or something similar.

Reply:
@kenlewis-sfdo Yeah that works for me. This is one of those scenarios where if bad json is passed in at this point we likely have a bigger problem elsewhere but an error would at least allow for quicker debugging instead of having to trace it down if the page just doesn't open. At this point, however, I wouldn't expect the format of the json to change.